traversal_test.go

```diff
@@ -86,10 +86,10 @@ func TestReferenceTraversal(t *testing.T) {
 
 		elements := 0
 		refs := 0
-		walkJSON(rs, func(elem JSONPather) error {
+		walkJSON(rs, func(elem interface{}) error {
 			elements++
-			if sch, ok := elem.(*Schema); ok {
-				if sch.HasKeyword("$ref") {
+			if kw, ok := elem.(SchemaKeyword); ok {
```

Comment:
The old behaviour doesn't work because you can't apply type-conversion from `*Not` (defined by `type Not Schema`) to `*Schema`, so it always returned false.